django/thunderstore/repository/api/v1/serializers.py

```diff
@@ -12,9 +12,16 @@ class PackageVersionSerializer(ModelSerializer):
     dependencies = SerializerMethodField()
 
     def get_download_url(self, instance):
-        url = instance.download_url
         if "request" in self.context:
-            url = self.context["request"].build_absolute_uri(instance.download_url)
+            url = self.context["request"].build_absolute_uri(
+                instance.download_url(
+                    self.context["view"].kwargs["community_identifier"]
+                )
+            )
+        else:
+            url = instance.download_url(
```

Comment:
Why is this in a if/else pattern this time around? Can't we use a slightly improved version of what was prior (set `url` at the start of the function, and do `url = self.context["request"].build_absolute_uri(url)` if request is available)?

Reply:
This way it has less variable assignments, so I think it's a bit more efficient? Then again yes, we can use what you said, I just somewhat defaulted into making it more efficient 🤷